scripts/ethEndToEndTest.js

```diff
@@ -0,0 +1,212 @@
+    console.log('Deploying ERC20Peg contract...');
+    const peg = await erc20Peg.deploy(bridge.address);
+    await peg.deployed();
+    console.log('CENNZnet erc20peg deployed to:', peg.address);
+
+    // Activate deposits
+    console.log(await peg.activateDeposits());
+    // Activate withdrawals
+    console.log(await peg.activateWithdrawals());
+
+    /***************** Ethereum Setup ends here **********************/
+
+    /***************** Make Deposit on ETHEREUM **********************/
+
+    const ethBalanceBefore = await ethers.provider.getBalance(peg.address);
+    console.log('Balance before::', ethBalanceBefore.toString());
+    const ethAddress = '0x0000000000000000000000000000000000000000';
+    let depositAmount = 5644;
+    // Alice
+    let cennznetAddress = "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d";
+    const deposition = await peg.deposit(ethAddress, depositAmount, cennznetAddress, {
+        gasLimit: 500000,
+        value: depositAmount
+    });
+    const depositTxHash = deposition.hash;
+    console.log('depositTxHash:::',depositTxHash);
+
+    const ethBalanceAfter = await ethers.provider.getBalance(peg.address);
+    console.log('Balance after::', ethBalanceAfter.toString());
+
+
+
+    /***************** CENNZnet Setup **********************/
+        // connecting to local chain
+    const api = await Api.create();
+    const keyring = new Keyring({type: 'sr25519'});
+    const alice = keyring.addFromUri('//Alice');
+
+    /********************* Need the following set up *****************************/
+    /********************* only for local chain **********************************/
+    /********************* wont need this for rata/nikau/azalea ******************/
+    const transaction1 = api.tx.erc20Peg.activateDeposits(true);
+    const transaction2 = api.tx.erc20Peg.activateWithdrawals(true);
+    const transaction3 = api.tx.erc20Peg.setContractAddress(peg.address);
+    const transaction4 = api.tx.ethBridge.setEventConfirmations(0); // Hardhat only makes blocks when txs are sent
+    const batchBridgeActivationEx = api.tx.utility.batch([
+        transaction1,
+        transaction2,
+        transaction3,
+        transaction4
+    ]);
+
+    await new Promise(async (resolve, reject) => {
+        await api.tx.sudo.sudo(batchBridgeActivationEx).signAndSend(alice, async ({status, events}) => {
+            if (status.isInBlock) {
+                events.forEach(({phase, event: {data, method, section}}) => {
+                    console.log('\t', phase.toString(), `: ${section}.${method}`, data.toString());
+                    resolve();
+                });
+            }
+        });
+    });
+    /********************* Set up complete *******************************************/
+
+    /********** new Token Id that would be minted on deposition at CENNZnet **********/
+    const testTokenId = await api.query.genericAsset.nextAssetId();
```

Comment:
could do query `api.query.erc20peg.erc20ToAssetId(ethAddress)` to check if it's mapped already

Reply:
done